### PR TITLE
Validate page and pageSize query parameters

### PR DESCRIPTION
**Pull request by BhuwanPandey**
For page and Pagesize query paramter , Set the default value when it contain non integer value .

Fix issue: #751 

Page and PageSize set to default value when called as 
https:domainname/admin/address/list?page=somevalue
https:domainname/admin/address/list?page=2&pageSize=somevalue

and
when called as  
https:domainname/admin/address/list?page=num 
https:domainname/admin/address/list?pageSize=num
where num lies in  (0,-1,-2...)
here, page and pagesize take only positive integer

### CODE REVIEW

**Comment by BhuwanPandey** [comment]
@aminalaee This PR is ready for review

File: sqladmin/models.py [hunk 1]
```diff
@@ -688,6 +688,8 @@ def __init__(self) -> None:
         self._custom_actions_in_detail: Dict[str, str] = {}
         self._custom_actions_confirmation: Dict[str, str] = {}
 
+        self.page_size_ = self.page_size
```
Comment (aminalaee) [inline sqladmin/models.py:691]: I think this is a bit unclear, maybe it's better to have a method `def get_page_size` instead?

File: sqladmin/models.py [hunk 2]
```diff
@@ -746,18 +748,29 @@ def _default_formatter(self, value: Any) -> Any:
 
         return value
 
+    def get_query_param(self, request: Request, key: str, default: int) -> int:
```
Comment (aminalaee) [inline sqladmin/models.py:751]: And also `validate_page_number` as it is not for any query parameter but integer only.

Comment (BhuwanPandey) [reply sqladmin/models.py:751]: you mean that get_query_param should be replaced with validate_page_number

Comment (aminalaee) [reply sqladmin/models.py:751]: Yeah that also works.

File: sqladmin/models.py (hunk 1 above)
Comment (aminalaee) [reply sqladmin/models.py:691]: I think you should get rid of `self._page_size` as we already have a `self.page_size`. In order to get rid of that I think a helper method will be useful.

Comment (BhuwanPandey) [reply sqladmin/models.py:691]: I think , defining any method,we need variable to track changeable value of pagesize.
but, What variable do you suggest for that?
In above scenario, by replaceing self._page_size with something else will resolve the case.

Comment (aminalaee) [reply sqladmin/models.py:691]: I don't think we have to store it in a variable? Just as it was previously working it's either the current value coming from the request parameter or the default self.page_size.

Comment (BhuwanPandey) [reply sqladmin/models.py:691]: In list.html line_number  (197)
 {{ request.query_params.get("pageSize") or model_view.page_size }} / Page
 
when we leave as it is,
on passing pagesize=somevalue
domainname/admin/address/list?pageSize=somevalue
 
 this somevalue is displayed like
 pagesize

Comment (aminalaee) [reply sqladmin/models.py:691]: yeah but you could call`{{ model_view.validate_page_size(request.query_params.get("pageSize")) or model_view.page_size }} / Page` right?

Comment (BhuwanPandey) [reply sqladmin/models.py:691]: Yes,
we can do that, as well we have to set default value and change it's structure like

 model_view.validate_page_size((request.query_params.get("pageSize")) or model_view.page_size),model_view.page_size)
 
but it looks quite messy

Comment (aminalaee) [reply sqladmin/models.py:691]: I think it's better than having `self.page_size` and `self.page_size_` and also the page size should be evaluated per request, we can't store the current page size on model_view. The `page_size` stored is the configuration which doesn't change per request, we shouldn't store per request page_size there.

Comment (BhuwanPandey) [reply sqladmin/models.py:691]: in list.html
          {{ model_view.validate_page_number((request.query_params.get("pageSize") or model_view.page_size),model_view.page_size)  }} / Page

in models.py
def validate_page_number(self, page_value: int|str, default: int) -> int:
        try:
            return int(page_value)
        except ValueError:
            return default
  
async def list(self, request: Request) -> Pagination:
        page = max(1,self.validate_page_number(request.query_params.get("page", 1), 1))
        page_size = self.validate_page_number(request.query_params.get("pageSize", 0), 0)
        
 After updating  , it looks like that
 If all the changes looks good to you
 I will commit the changes

**Comment by BhuwanPandey** [comment]
@aminalaee  Again, It is ready for Review

**Comment by aminalaee** [comment]
If I understand correctly you expect these URLs to fail. This PR is not deployed yet, page=-500 will throw 400 exception.

**Comment by BhuwanPandey** [comment]
> If I understand correctly you expect these URLs to fail. This PR is not deployed yet, page=-500 will throw 400 exception.

  when i tested locally, this condition will not trigger either page = -500 or pagesize = -1
  but, when both negative value of page and pagesize passed at once this condition is trigger...
  
  application.py
      if (
            pagination.page * pagination.page_size
            > pagination.count + pagination.page_size
        ):
            raise HTTPException(
                status_code=400, detail="Invalid page or pageSize parameter"
            )

**Review by aminalaee — APPROVED** [review]
Thank you for the contribution 👍

**Pull request merged**